### PR TITLE
[risk=low] DB-246 survey actual question number display and order change

### PR DESCRIPTION
**Pull request by SrushtiGangireddy**
1. Display the surveys in the order that user takes them. (Fetched this change from other repo).
2. Display the survey question number as the actual order number in which they took the survey.

### CODE REVIEW

File: common-api/src/main/java/org/pmiops/workbench/cdr/dao/SurveyModuleDao.java [hunk 1]
```diff
@@ -14,22 +14,23 @@
...
-      "group by m.name, m.description, m.concept_id\n" +
-      "order by m.name")
+          "m.concept_id, COUNT(DISTINCT c.concept_id) as question_count,\n" +
+          // We don't show participant counts when filtering by keyword, and don't have a way of computing them easily; return 0.
+          "0 participant_count, m.order_number \n" +
+          "from survey_module m\n" +
+          "join achilles_results r on m.concept_id = r.stratum_1\n" +
+          "join concept c on r.stratum_2 = c.concept_id\n" +
+          "join survey_question_map  sqm on sqm.question_concept_id = r.stratum_2\n" +
+          "where r.analysis_id = 3110 and sqm.sub=0\n" +
```
Comment (SrushtiGangireddy) [inline common-api/src/main/java/org/pmiops/workbench/cdr/dao/SurveyModuleDao.java:24]: Only fetching count of matching main questions (not counting sub questions)

File: common-api/src/main/java/org/pmiops/workbench/cdr/dao/SurveyModuleDao.java [hunk 2]
```diff
@@ -14,22 +14,23 @@
...
+          "and (match(c.concept_name) against (?1 in boolean mode) > 0 or\n" +
+          "match(r.stratum_4) against(?1 in boolean mode) > 0) \n" +
+          "group by m.name, m.description, m.concept_id\n" +
+          "order by m.order_number")
   List<SurveyModule> findSurveyModuleQuestionCounts(String matchExpression);
 
   SurveyModule findByConceptId(long conceptId);
 
-  List<SurveyModule> findByOrderByName();
+  List<SurveyModule> findByOrderByOrderNumberAsc();
```
Comment (SrushtiGangireddy) [inline common-api/src/main/java/org/pmiops/workbench/cdr/dao/SurveyModuleDao.java:35]: Cloned this from other repo.

File: common-api/src/main/java/org/pmiops/workbench/cdr/model/SurveyModule.java [hunk 3]
```diff
@@ -13,19 +13,21 @@
...
-              .questionCount(surveyModule.getQuestionCount())
-              .participantCount(surveyModule.getParticipantCount());
+          (surveyModule) ->
+                  new org.pmiops.workbench.model.SurveyModule()
+                          .conceptId(surveyModule.getConceptId())
+                          .name(surveyModule.getName())
+                          .description(surveyModule.getDescription())
+                          .questionCount(surveyModule.getQuestionCount())
+                          .participantCount(surveyModule.getParticipantCount())
+                          .orderNumber(surveyModule.getOrderNumber());
```
Comment (SrushtiGangireddy) [inline common-api/src/main/java/org/pmiops/workbench/cdr/model/SurveyModule.java:23]: Clone change.

File: common-api/src/main/java/org/pmiops/workbench/cdr/model/SurveyModule.java [hunk 4]
```diff
@@ -13,19 +13,21 @@
...
 
   private long conceptId;
   private String name;
   private String description;
   private long questionCount;
   private long participantCount;
+  private int orderNumber;
```
Comment (SrushtiGangireddy) [inline common-api/src/main/java/org/pmiops/workbench/cdr/model/SurveyModule.java:30]: clone changes.

File: public-api/db-cdr/csv/survey_question_map.csv [hunk 5]
```diff
@@ -61,7 +61,7 @@ id,survey_concept_id,question_concept_id,survey_order_number,question_order_numb
 71,1585855,1586182,3,4,"1586182",0
 72,1585855,1586185,3,4,"1586182.1586183.1586185",1
 73,1585855,1586190,3,5,"1586190",0
-74,1585855,1586193,3,5,"1586190.1586191.1586193",0
+74,1585855,1586193,3,5,"1586190.1586191.1586193",1
```
Comment (SrushtiGangireddy) [inline public-api/db-cdr/csv/survey_question_map.csv:64]: bug fix.

File: public-api/db-cdr/generate-cdr/make-bq-data.sh [hunk 6]
```diff
@@ -578,7 +578,7 @@ set sm.question_count=num_questions from
     on r.stratum_1 = CAST(sq.survey_concept_id AS STRING)
   join \`${OUTPUT_PROJECT}.${OUTPUT_DATASET}.concept\` qc
     on sq.question_concept_id = qc.concept_id
-where r.analysis_id = 3110 and qc.count_value > 0
+where r.analysis_id = 3110 and qc.count_value > 0 and sq.sub=0
```
Comment (SrushtiGangireddy) [inline public-api/db-cdr/generate-cdr/make-bq-data.sh:581]: fetching the count of only main questions.

File: public-api/src/main/java/org/pmiops/workbench/cdr/dao/QuestionConceptDao.java [hunk 7]
```diff
@@ -10,7 +10,9 @@
     @Query(nativeQuery=true, value="SELECT c.concept_id,c.concept_name,c.domain_id,c.vocabulary_id,c.concept_code,c.count_value,c.prevalence from\n" +
             "concept c \n" +
             "join achilles_results ar on ar.stratum_2=c.concept_id\n" +
+            "join survey_question_map sqm on sqm.question_concept_id=ar.stratum_2\n" +
```
Comment (SrushtiGangireddy) [inline public-api/src/main/java/org/pmiops/workbench/cdr/dao/QuestionConceptDao.java:13]: fetching only main questions.

File: public-api/src/main/java/org/pmiops/workbench/cdr/model/QuestionConcept.java [hunk 8]
```diff
@@ -19,6 +19,7 @@
     private AchillesAnalysis genderAnalysis;
     private AchillesAnalysis ageAnalysis;
     private AchillesAnalysis genderIdentityAnalysis;
+    private List<SurveyQuestionMap> questions = new ArrayList<>();
```
Comment (SrushtiGangireddy) [inline public-api/src/main/java/org/pmiops/workbench/cdr/model/QuestionConcept.java:22]: Joined to get the path from survey question map row.

**Review by mhigbyflowers — APPROVED** [review]
looks great.
Thanks for all of your comments.

**Pull request merged**